### PR TITLE
BZ-1870696: Adding a basic landing page for the security book

### PR DESCRIPTION
**Pull request by bergerhoffer**
https://bugzilla.redhat.com/show_bug.cgi?id=1870696

Preview: https://deploy-preview-35208--osdocs.netlify.app/openshift-enterprise/latest/security/index.html

Added a landing page for the "Security and compliance" book. Also reorganized the subfolders into a more logical order. 

I also moved the oc-compliance plugin assembly in with the compliance operator subfolder and renamed the subfolder to just "Compliance". Will need to get an ack on whether this is okay or not, and if so, if it needs a redirect.

Also addresses #27767

### CODE REVIEW

File: security/index.adoc
Comment (TanyaShearonRH) [inline security/index.adoc:15]: I might move this first sentence to the Container security section: "It is important to understand how to properly secure various aspects of your OpenShift Continer Platform cluster" since there is only one sentence in the Security overview section.

File: security/index.adoc
Comment (TanyaShearonRH) [inline security/index.adoc:75]: I wonder if the text in the "Compliance overview" and "Compliance checking" sections could be combined within a single heading.

File: security/index.adoc
Comment (TanyaShearonRH) [inline security/index.adoc:86]: For Additional resources, the first link was referenced earlier in the topic. Do we want to point to the same reference material twice?

File: security/index.adoc
Comment (TanyaShearonRH) [inline security/index.adoc:34]: The sentence starts as "Audit provides a security-relevant..." Should "Audit" be changed to Auditing" or something else?

File: security/index.adoc
Comment (TanyaShearonRH) [inline security/index.adoc:90]: LGTM :-)

File: security/index.adoc
Comment (bergerhoffer) [reply security/index.adoc:34]: Yeah, I pulled that wording from another file in the existing docs. I thought it sounded weird too. I don't think it should be "Auditing" though, so I'll have to give it some thought

File: security/index.adoc
Comment (bergerhoffer) [reply security/index.adoc:86]: @TanyaShearonRH Can you specify which link you're talking about? If it's this "Understanding authentication" link, then I think this is the only place we mention it.

File: security/index.adoc
Comment (bergerhoffer) [reply security/index.adoc:75]: So under "Compliance overview", there are two subsections, "Compliance checking" and "File integrity checking". I've categorized the file integrity section under the compliance umbrella. But if that's not the case and it needs to be broken out into its own top-level section, then in that case I could combine the two compliance headings.

File: security/index.adoc
Comment (bergerhoffer) [reply security/index.adoc:15]: @TanyaShearonRH I know that this could use beefed up (and also need an intro blurb at the top of the assembly). But usually I try to avoid having back-to-back headings with no text underneath a heading.

File: security/index.adoc
Comment (bergerhoffer) [reply security/index.adoc]: @TanyaShearonRH Okay so I changed it to "{product-title} auditing" (will resolve to "OpenShift Container Platform auditing")

File: _topic_map.yml
Comment (bergerhoffer) [inline _topic_map.yml:750]: @ahardin-rh Can you please take a look at the change I made here?

I moved the oc-compliance plug-in assembly in with the compliance operator docs (since it's directly related to it), and then renamed the overall folder to "Compliance".

**Review by JAORMX — APPROVED** [review]

File: _topic_map.yml
Comment (bergerhoffer) [reply _topic_map.yml:750]: Excellent, thank you!

File: security/index.adoc
Comment (bergerhoffer) [inline security/index.adoc:76]: @vikram-redhat with this PR, I'm moving the oc-compliance plug-in assembly. I can't remember where we left off with requiring redirects for every move or if we weren't adding them for everything now. thoughts?

**Comment by bergerhoffer** [comment]
Moving forward with this as is for now. Left a few TODO items in the source for someone else who has been working with the overview/landing pages initiative to potentially look into building out a little more.

And just need to check in on whether a redirect is required before merging.

Note to self: Make sure to strip out anything that isn't applicable when backporting.

File: security/index.adoc
Comment (adellape) [inline security/index.adoc:18]: IBM SG would suggest something like:

```
These sections also include information on the following topics:
```

to avoid continuing the lead-in sentence into a list. Also dropping the punctuation from all the list items since they're fragments.

File: security/index.adoc
Comment (adellape) [inline security/index.adoc:32]: Loving these assembly-level xrefs. :sunglasses:

File: security/index.adoc
Comment (adellape) [inline security/index.adoc:67]: s/industry standards/industry standards,/

File: security/index.adoc
Comment (adellape) [inline security/index.adoc:15]: Since anchors will still be autogenerated for these even though they're `[discrete]`, suggest adding unique IDs to them all.

**Review by adellape — COMMENTED** [review]
Looks good, only a few thoughts.

File: security/index.adoc
Comment (bergerhoffer) [reply security/index.adoc:15]: You know, I never really thought about that. And look at that, it's already in the guidelines with an example showing the discrete heading w/ an ID still. Thanks for pointing that out, I'll have to fix this in some of my other content sometime.

**Comment by bergerhoffer** [comment]
@xingxingxia This update is adding a landing page for the "Security and compliance" section. It's basically just a quick overview of the various security topics we have and then linking out to them. A good bit of the wording has been taken from the various sections. Can you please review (or pull in whoever else we need to review)?

Preview: https://deploy-preview-35208--osdocs.netlify.app/openshift-enterprise/latest/security/index.html

**Comment by xingxingxia** [comment]
@bergerhoffer , you can ask @xiaojiey to review who is the subteam lead of security and compliance subteam.
Though, I reviewed through the doc, LGTM, good index page! BTW, this index page does not mention TLS profile and JavaScript, does it need to mention?
@xiaojiey , I changed the bug QA Contact to you. Please review if you have more comment.

File: _topic_map.yml
Comment (xiaojiey) [inline _topic_map.yml:729]: Should it be Compliance Operator?

Comment (xingxingxia) [reply _topic_map.yml:729]: > I also moved the oc-compliance plugin assembly in with the compliance operator subfolder and renamed the subfolder to just "Compliance". Will need to get an ack on whether this is okay or not, and if so, if it needs a redirect

This is PR top comment. To clarify: so you have concern about the moving and renaming?

Comment (xiaojiey) [reply _topic_map.yml:729]: "compliance" is a very big concept. It usually contains two part: "compliance checking" and "file integrity checking".
To be more accurate, maybe use "compliance-operator and oc-compliance plugin" instead.

Comment (bergerhoffer) [reply _topic_map.yml:729]: No worries, I can revert this change and update the folder name back to "Compliance Operator". That folder will still hold the oc-compliance plug-in doc though. But since it's related to the Compliance Operator, I think this is fine.

CC @JAORMX

**Comment by bergerhoffer** [comment]
@xiaojiey I updated the folder name from just "Compliance" back to "Compliance Operator". Can you please re-review and let me know if you're okay with these updates? Thanks!

**Review by JAORMX — APPROVED** [review]

**Comment by bergerhoffer** [comment]
@vikram-redhat This is ready to merge, if you can please set up a redirect for the oc-compliance assembly, for 4.8 only. That assembly doesn't exist in 4.6 or 4.7.

I will take care of the CPs to 4.6 and 4.7, because I might need to make adjustments to the landing page if the content is different in earlier versions.

**Pull request merged**
